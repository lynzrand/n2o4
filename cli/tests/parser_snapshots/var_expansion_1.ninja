# From: Ninja doc, "Lexical Syntax"

rule demo
  command = echo "this is a demo of $foo"

build out: demo
  foo = bar
