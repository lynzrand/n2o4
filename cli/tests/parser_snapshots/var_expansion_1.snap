Ok(
    NinjaFile {
        global_scope: {},
        rules: {
            "demo": Rule {
                vars: {
                    "command": Expandable(
                        [
                            Regular(
                                "echo \"this is a demo of ",
                            ),
                            Var(
                                "foo",
                            ),
                            Regular(
                                "\"",
                            ),
                        ],
                    ),
                },
            },
        },
        builds: [
            Build {
                inputs: [],
                implicit_inputs: [],
                order_only_inputs: [],
                outputs: [
                    "out",
                ],
                command: "echo \"this is a demo of bar\"",
                depfile: None,
                deps: None,
                msvc_deps_prefix: None,
                description: None,
                dyndep: None,
                generator: false,
                restat: false,
                rspfile: None,
                rspfile_content: None,
            },
        ],
    },
)
